syntax = "proto3";

package app;

enum Instrument {
  INSTRUMENT_UNSPECIFIED = 0;
  INSTRUMENT_SPOT = 1;
  INSTRUMENT_MARGIN = 2;
  INSTRUMENT_PERP = 3;
  INSTRUMENT_INVERSE = 4;
  INSTRUMENT_FUTURES = 5;
  INSTRUMENT_OPTION = 6;
}

enum Exchange {
  EXCHANGE_UNSPECIFIED = 0;
  EXCHANGE_BINANCE = 1;
  EXCHANGE_BINANCE_PERP = 2;
  EXCHANGE_BYBIT = 3;
}

enum Side {
  SIDE_UNSPECIFIED = 0;
  SIDE_BUY = 1;
  SIDE_SELL = 2;
}

enum TimeInForce {
  TIME_IN_FORCE_UNSPECIFIED = 0;
  TIME_IN_FORCE_GTC = 1;
  TIME_IN_FORCE_IOC = 2;
  TIME_IN_FORCE_FOK = 3;
}

enum OrderType {
  ORDER_TYPE_UNSPECIFIED = 0;
  ORDER_TYPE_LIMIT = 1;
  ORDER_TYPE_MARKET = 2;
  ORDER_TYPE_STOP_MARKET = 3;
}

message Symbol {
  string base = 1;
  string quote = 2;
}

